app/Models/Concerns/HasMemberships.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+namespace App\Models\Concerns;
+
+use App\Exceptions\Memberships\MembershipNotSecondaryException;
+use App\Exceptions\Memberships\PrimaryMembershipDoesntAllowSecondaryException;
+use App\Models\Membership;
+use App\Models\Membership\MembershipPivot;
+use Carbon\Carbon;
+use Illuminate\Database\Eloquent\Relations\BelongsToMany;
+
+trait HasMemberships
+{
+    private $primaryMembership;
+
+    public function memberships(): BelongsToMany
+    {
+        return $this->membershipsRelationship()
+            ->wherePivot('ended_at', null)
+            ->orderByPriority();
+    }
+
+    public function membershipHistory(): BelongsToMany
+    {
+        return $this->membershipsRelationship()
+            ->orderBy((new MembershipPivot())->getTable().'.started_at', 'desc');
```

Comment:
This feels odd.

Reply:
This was the best way I could come up with to not have to duplicate the table name lots of times. It works pretty well and I think it is pretty explicit